ash/src/device.rs

```diff
@@ -6,7 +6,19 @@ use std::mem;
 use std::os::raw::c_void;
 use std::ptr;
 
-/// <https://www.khronos.org/registry/vulkan/specs/1.3-extensions/man/html/VkDevice.html>
+/// This is a dispatchable object, whose main purpose is to provide access to Vulkan's functions.
+///
+/// It is not identical to Vulkan's device object (see [vk::Device]).
+/// The latter, however, can be accessed via the [`handle()`][Self::handle()] method.
+/// In this sense, `ash::Device` is a Vulkan device handle together with all its associated function pointers.
+///
+/// All functions from the Vulkan API (except those from extensions) which have a [`VkDevice`] as their first argument, have become methods of `ash::Device`.
+/// Their [`VkDevice`] argument is always passed implicitly.
```

Comment:
This is not entirely correct. First argument can also be a derivative from the current device (ie. associated with one unique device) such as a `CommandBuffer`, `PhysicalDevice`, `Queue` etc. I'd argue just having the paragraph above with the `device function pointers loaded from it` mention is enough, unless you feel like linking to the Vulkan object model docs and mentioning this device relationship is necessary.

Reply:
Also not sure if referencing `VkDevice` from the Vulkan docs is very helpful, pointing to the Rust version here is more type-correct (only reference Vulkan docs if there's something actually useful to be read there, and be explicit about it providing additional information).

And in any case, since this first argument is passed implicitly internally I don't think we should bother the user with that detail - but if you feel like this being necessary, expand on that in the `providing convenient wrappers to call these functions` sentence.

Reply:
> This is not entirely correct. First argument can also be a derivative from the current device
I'd say that does not contradict the sentence which only talks about the functions with VkDevice as first argument. (It does make my description incomplete; and I have removed this sentence.)

About "bothering the user": You can see this as an internal implementation detail of ash — or as one of the most striking differences between ash and the Vulkan docs to which we often refer, which could (should?) be made explicit somewhere.